lldb/source/Target/Thread.cpp

```diff
@@ -813,12 +819,17 @@ bool Thread::ShouldStop(Event *event_ptr) {
   // decide whether they still need to do more work.
 
   bool done_processing_current_plan = false;
-
   if (!current_plan->PlanExplainsStop(event_ptr)) {
     if (current_plan->TracerExplainsStop()) {
       done_processing_current_plan = true;
       should_stop = false;
     } else {
+      // Leaf plan that does not explain the stop should be popped.
+      // The plan should be push itself later again before resuming to stay
+      // as leaf.
+      if (current_plan->IsLeafPlan() && current_plan->MischiefManaged())
```

Comment:
If the plan is a leaf plan and doesn't explain the stop, it should always be popped, shouldn't it?  Why do you include the MischiefManaged check here?

Reply:
Thanks. I thought I had a reason to add this check 3 weeks ago but could not remember anymore. Running all the tests without the check pass so it might be a legacy change. Removed.